examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+	wantedCategories := []string{
+		"config.hardware.numCPU",
+		"config.hardware.memoryMB",
+		"config.hardware.numCoresPerSocket",
+		"config.memoryHotAddEnabled",
+		"config.cpuHotRemoveEnabled",
+		"config.cpuHotAddEnabled",
+	}
+
+	cfgs := []vmConfig{}
+
+	for _, attTag := range attachedTags {
+		catObj, err := c.tagMgr.GetCategory(ctx, attTag.CategoryID)
+		if err != nil {
+			return []vmConfig{}, fmt.Errorf("get category from catID: %w", err)
+		}
+
+		for _, want := range wantedCategories {
+			if catObj.Name == want {
+				// Remove the 'config.hardware.' prefix.
+				cfgType := strings.TrimPrefix(catObj.Name, "config.")
+				cfgType = strings.TrimPrefix(cfgType, "hardware.")
+
+				cfg, err := buildConfig(attTag, cfgType)
+				if err != nil {
+					return []vmConfig{}, fmt.Errorf("build config: %w", err)
+				}
+
+				cfgs = append(cfgs, cfg)
+			}
+		}
+	}
+
+	return cfgs, nil
+}
+
+// buildConfig will get the config value from an attached tag. Then, it will return
+// a VM config containing the config type and value.
+func buildConfig(attachedTag tags.Tag, cfgType string) (vmConfig, error) {
+	if cfgType == "numCPU" || cfgType == "memoryMB" || cfgType == "numCoresPerSocket" {
+		val, err := strconv.Atoi(attachedTag.Name)
+		if err != nil {
+			return vmConfig{}, fmt.Errorf("convert string to int :%w", err)
+		}
+
+		return vmConfig{name: cfgType, valInt: val}, nil
+	}
+
+	if cfgType == "memoryHotAddEnabled" || cfgType == "cpuHotRemoveEnabled" || cfgType == "cpuHotAddEnabled" {
+		val, err := strconv.ParseBool(attachedTag.Name)
+		if err != nil {
+			return vmConfig{}, fmt.Errorf("convert string to bool: %w", err)
+		}
+
+		return vmConfig{name: cfgType, valBool: val}, nil
+	}
+
+	return vmConfig{}, errors.New(cfgType + " is not an expected config type")
+}
+
+// filterOutCurrentConfigs removes the tag configs that are already current configs.
+func filterOutCurrentConfigs(tagConfigs, currConfigs []vmConfig) []vmConfig {
+	unappliedCfgs := []vmConfig{}
+
+	for _, tc := range tagConfigs {
+		if !isCfgCurrent(tc, currConfigs) {
+			unappliedCfgs = append(unappliedCfgs, tc)
+		}
+	}
+	return unappliedCfgs
+}
+
+// isHwCfgMatch determines if the given tag's hardware configuration
+// is already the current configuration of the hardware.
+func isCfgCurrent(tagConfig vmConfig, currConfigs []vmConfig) bool {
+	for _, curr := range currConfigs {
+		if tagConfig == curr {
+			return true
+		}
+	}
+
+	return false
+}
+
+// makeCfgsMatch sets configuration of the VM to that of the attached tag.
+func (c *vsClient) applyCfgs(ctx context.Context, moVM mo.VirtualMachine, mor types.ManagedObjectReference, cfgs []vmConfig) (*object.Task, error) {
+	vm := object.NewVirtualMachine(c.govmomi.Client, mor)
+	desiredSpec := generateDesiredSpec(cfgs, moVM)
+
+	task, err := vm.Reconfigure(ctx, desiredSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	if debug {
+		log.Println("VM reconfigured with", desiredSpec)
```

Comment:
Missing spaces between strings.

Reply:
Println adds spaces in between the arguments.